jkube-kit/enricher/generic/src/main/java/org/eclipse/jkube/enricher/generic/SecretEnricher.java

```diff
@@ -129,10 +131,7 @@ private void addSecretsFromXmlConfiguration(KubernetesListBuilder builder) {
 
     private List<SecretConfig> getSecretsFromXmlConfig() {
         ResourceConfig resourceConfig = getConfiguration().getResource();
-        if(resourceConfig != null && resourceConfig.getSecrets() != null) {
-            return resourceConfig.getSecrets();
-        }
-        return null;
+        return Optional.ofNullable(resourceConfig).map(resourceConfig1 -> Optional.ofNullable(resourceConfig1.getSecrets()).orElse(Collections.emptyList())).orElse(Collections.emptyList());
```

Comment:
minor suggestion regarding formatting
```suggestion
        return Optional.ofNullable(getConfiguration().getResource())
            .map(rc -> Optional.ofNullable(rc.getSecrets())
                .orElse(Collections.emptyList()))
            .orElse(Collections.emptyList());
```